File: flutterApp/flutter_application_1/lib/models/feedback.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'feedback.freezed.dart';
part 'feedback.g.dart';

@freezed
class Feedback with _$Feedback {
  const factory Feedback({
    String? id,
    required double rating,
    required String service,
    required String title,
    required String content,
    String? pros,
    String? cons,
    String? suggestions,
    String? email,
    String? phone,
    List<String>? images,
    String? userId,
    DateTime? createdAt,
  }) = _Feedback;

  factory Feedback.fromJson(Map<String, dynamic> json) => _$FeedbackFromJson(json);
} 

File: database/models/Feedback.js
import mongoose from 'mongoose';

const feedbackSchema = new mongoose.Schema({
  title: { type: String, required: true },
  content: { type: String, required: true },
  rating: { type: Number, required: true, min: 1, max: 5 },
  service: { type: String, required: true, enum: ['ai', 'online', 'appointment', 'other'] },
  pros: { type: String },
  cons: { type: String },
  suggestions: { type: String },
  email: { type: String },
  phone: { type: String },
  images: [{ type: String }],
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  createdAt: { type: Date, default: Date.now }
});

export default mongoose.model('Feedback', feedbackSchema); 